src/migrations/20220819014844-user-settings-table.js

```diff
@@ -0,0 +1,85 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => queryInterface.sequelize.transaction(async (transaction) => {
+    await queryInterface.createTable(
+      'UserSettings',
+      {
+        id: {
+          type: Sequelize.INTEGER,
+          primaryKey: true,
+          autoIncrement: true,
+          allowNull: false,
+        },
+        // 'email', etc.
+        class: { type: Sequelize.STRING, allowNull: false },
+        key: { type: Sequelize.STRING, allowNull: false },
+        default: { type: Sequelize.STRING, allowNull: false },
+        createdAt: { allowNull: false, type: Sequelize.DATE },
+        updatedAt: { allowNull: false, type: Sequelize.DATE },
+      },
+      { transaction },
+    );
+
+    const keys = [
+      'emailWhenReportSubmittedForReview',
+      'emailWhenChangeRequested',
+      'emailWhenReportApproval',
+      'emailWhenAppointedCollaborator',
+    ];
+
+    await Promise.all(keys.map(async (key) => {
+      await queryInterface.sequelize.query(
+        `INSERT INTO "UserSettings" ("class", "key", "default", "createdAt", "updatedAt") VALUES ('email', '${key}', 'never', current_timestamp, current_timestamp)`,
+        { transaction },
+      );
+    }));
+
+    await queryInterface.createTable(
+      'UserSettingOverrides',
+      {
+        id: {
+          type: Sequelize.INTEGER,
+          allowNull: false,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        userId: {
+          allowNull: false,
+          type: Sequelize.INTEGER,
+          references: { model: { tableName: 'Users' }, key: 'id' },
+        },
+        userSettingId: {
+          allowNull: false,
+          type: Sequelize.INTEGER,
+          references: { model: { tableName: 'UserSettings' }, key: 'id' },
+        },
+        value: { allowNull: false, type: Sequelize.STRING },
```

Comment:
can we change this to a jsonb datatype, this would allow complex settings to be saved, Like saving filters

Reply:
5fbf645